codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +275,12 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                String http1Request = in.toString(0, 1024, CharsetUtil.US_ASCII);
+                int http1Index;
+                if ((http1Index = http1Request.indexOf("HTTP/1.")) != -1) {
+                    throw connectionError(PROTOCOL_ERROR, "Unexpected HTTP/1 request: %s",
```

Comment:
And I also wonder if we really should include these 1024 bytes string here as it may contain "sensitive" data ?